### PR TITLE
sql/migrate: add template formatter

### PR DESCRIPTION
**Pull request by masseelch**
This PR adds `TemplateFormatter` which implements `Formatter` and uses `text/template.Template`s to define how migrations files are written.

A default formatter is given, that is compatible with [golang-migrate/migrate](https://github.com/golang-migrate/migrate).

### CODE REVIEW

File: sql/migrate/migrate.go [hunk 1]
```diff
@@ -222,3 +203,122 @@ func (p *Planner) WritePlan(plan *Plan) error {
...
+			},
+		},
+	}
+)
+
+// NewTemplateFormatter creates a new Formatter working with the given templates.
+//
+//	migrate.NewTemplateFormatter(
+//		template.Must(template.New("").Parse("{{now.Unix}}{{.Name}}.sql")),					// name template
+//		template.Must(template.New("").Parse("{{range .Changes}}{{println .Cmd}}{{end}}")),	// content template
```
Comment (rotemtam) [inline sql/migrate/migrate.go:268]: comment seems too indented on L267

**Review by rotemtam — APPROVED** [review]

File: sql/migrate/migrate.go [hunk 2]
```diff
@@ -222,3 +203,122 @@ func (p *Planner) WritePlan(plan *Plan) error {
...
 	return nil
 }
+
+// LocalDir implements Dir for a local path.
+type LocalDir struct {
+	dir string
+}
+
+// Open implements Dir.Open.
+func (dir *LocalDir) Open(name string) (File, error) {
```
Comment (a8m) [inline sql/migrate/migrate.go:213]: use short var.
```suggestion
func (d *LocalDir) Open(name string) (File, error) {
```

File: sql/migrate/migrate.go [hunk 3]
```diff
@@ -222,3 +203,122 @@ func (p *Planner) WritePlan(plan *Plan) error {
...
+	t := new(TemplateFormatter)
+	for i := 0; i < len(templates); i += 2 {
+		t.templates = append(t.templates, struct{ N, C *template.Template }{templates[i], templates[i+1]})
+	}
+	return t, nil
+}
+
+// Format implements the Formatter interface.
+func (t *TemplateFormatter) Format(plan *Plan) ([]File, error) {
+	var fs []File
```
Comment (a8m) [inline sql/migrate/migrate.go:284]:
```suggestion
	fs := make([]File, 0, len(t.templates))
```

**Review by a8m — APPROVED** [review]
Looks good. The name `TemplateFormatter` feels a big long for me, but I still like it 😃

**Pull request merged**